workspaces/cli-server/src/sentry/index.ts

```diff
@@ -0,0 +1,33 @@
+import * as Sentry from '@sentry/node';
+
+export interface ICliServerSentryWrapper {
+  init(): void;
+}
+
+class CliServerSentryWrapper implements ICliServerSentryWrapper {
+  init() {
+    Sentry.init({
+      dsn: process.env.OPTIC__CLI_SERVER__SENTRY__DSN,
+      serverName: 'optic-cli-server',
+      environment:
+        process.env.OPTIC__CLI_SERVER__SENTRY__ENVIRONMENT || 'local',
+      release: process.env.OPTIC__NPM__VERSION || 'local',
+      tracesSampleRate: 1.0,
+    });
+    console.log('Sentry is enabled');
+  }
+}
```

Comment:
Parsing environment variables to establish a config doesn't belong this deep into the codebase, but way closer to the edge of the program. Anything that's outside a `main` function or global `config` (that's immediately consumed by that `main` function) should deal with plain objects and values. Not doing so can make it incredibly opaque to reason about why code behaves the way that it does, hard to re-use anything, etc.

Once you take this logic out of here, I see little reason for why this needs to be an interface of a class and 2 implementations of it: a function that registers sentry based on whether a dsn was passed should be more than enough. Couple of lines, easier to parse and understand.

Reply:
Sentry is meant to be loaded as early as possible from the main module like you say about configs, but I agree the envs can move to config so this module uses plain values. The interface is a remnant of me attempting to support the sentry express integration, which had more methods after init. I still think it is a good idea to create the separation for any 3rd party dependency and is overall simpler to maintain as an instance that callers control rather than a function with conditional effects

Reply:
> The interface is a remnant of me attempting to support the sentry express integration, which had more methods after init. 

@devdoshi yeah, it seemed like something like that.

> I still think it is a good idea to create the separation for any 3rd party dependency and is overall simpler to maintain as an instance that callers control rather than a function with conditional effects.

Totally, I've been in projects where there were multiple remote-error tracking integrations at once, where we soon ended up with a generic `errorTracker` interface. Since it's mostly glue code that wasn't too hard to retrofit.